account_cryptocurrency/models/res_currency_move.py

```diff
@@ -0,0 +1,199 @@
+                                 string='Direction', required=True,
+                                 readonly=True,
+                                 states={'draft': [('readonly', False)]},
+                                 )
+    move_line_ids = fields.One2many('res.currency.move.line', 'move_id',
+                                    string='Currency move lines',
+                                    copy=False,
+                                    readonly=True,
+                                    states={'draft': [('readonly', False)]},
+                                    )
+
+    @api.depends('currency_id', 'journal_id', 'direction')
+    def _compute_accounts(self):
+        for rec in self:
+            inventory_account = self.currency_id.with_context(
+                force_company=self.company_id.id).inventory_account_id
+
+            if self.direction == 'inbound':
+                rec.debit_account_id = inventory_account
+                rec.credit_account_id = \
+                    self.journal_id.default_credit_account_id
+            else:
+                rec.debit_account_id = \
+                    self.journal_id.default_debit_account_id
+                rec.credit_account_id = inventory_account
+
+    def _get_sequence(self):
+        return 'res.currency.move'
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('name'):
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                self._get_sequence()) or _('New')
+        return super(ResCurrencyMove, self).create(vals)
+
+    def _prepare_incoming_move_line(self):
+        self.ensure_one()
+        amount = self.currency_id.with_context(date=self.date).compute(
+            self.amount, self.company_id.currency_id)
+        return {
+            'move_id': self.id,
+            'quantity': self.amount,
+            'date': self.date,
+            'amount': amount,
+        }
+
+    def _prepare_outgoing_move_line(self, candidate, qty):
+        self.ensure_one()
+        return {
+            'move_id': self.id,
+            'quantity': qty,
+            'date': self.date,
+            'in_move_line_id': candidate.id,
+            'amount': candidate.price_unit * qty
+        }
+
+    def _get_fifo_candidates_in_move_line(self):
+        """ Find IN moves that can be used to value OUT moves.
+        """
+        self.ensure_one()
+        domain = [('currency_id', '=', self.currency_id.id),
+                  ('remaining_qty', '>', 0.0)] + self.env[
+            'res.currency.move.line']._get_in_base_domain(
+            company_id=self.company_id.id)
+        candidates = self.env['res.currency.move.line'].search(
+            domain, order='date, id')
+        return candidates
+
+    def _run_fifo(self):
+        self.ensure_one()
+        qty_to_take_on_candidates = self.amount
+        candidates = self._get_fifo_candidates_in_move_line()
+        for candidate in candidates:
```

Comment:
What about `for candidate in candidates.filtered('remaining_qty'):`?
Using this way you reduce the cyclomatic complexity and then reduces the qty of tests to cover this line